### PR TITLE
add laplacian_spectrum example

### PR DESCRIPTION
**Pull request by paulitapb**
Added an example where the laplacian_spectrum is used to get the number of eigenvalues of the Laplacian matrix that are zero to count the number of connected components of a graph. 

Also added a missed link to to_numpy_array

### CODE REVIEW

File: networkx/linalg/spectrum.py [hunk 1]
```diff
@@ -32,11 +32,23 @@ def laplacian_spectrum(G, weight="weight"):
...
-    See to_numpy_array for other options.
+    See :func:`~networkx.convert_matrix.to_numpy_array` for other options.
 
     See Also
     --------
     laplacian_matrix
+
+    Examples
+    --------
+    The multiplicity of O as an eigenvalue of the laplacian matrix is equal
```
Comment (rossbar) [inline networkx/linalg/spectrum.py:43]:
```suggestion
    The multiplicity of 0 as an eigenvalue of the laplacian matrix is equal
```

File: networkx/linalg/spectrum.py [hunk 2]
```diff
@@ -32,11 +32,23 @@ def laplacian_spectrum(G, weight="weight"):
...
+    to the number of connected components of G.
+
+    >>> import numpy as np
+    >>> G = nx.graph_atlas(22) #This graph from the Graph Atlas has 3 connected components.
```
Comment (rossbar) [inline networkx/linalg/spectrum.py:47]: IMO it'd make the example more clear if the reader knew what the graph was (I doubt many have the graph atlas memorized :joy: ):

```suggestion
    >>> G = nx.Graph()  # Create a graph with 5 nodes and 3 connected components
    >>> G.add_nodes_from(range(5))
    >>> G.add_edges_from([(0, 2), (3, 4)])
```

File: networkx/linalg/spectrum.py [hunk 3]
```diff
@@ -32,11 +32,23 @@ def laplacian_spectrum(G, weight="weight"):
...
+    >>> eigvals = nx.laplacian_spectrum(G).round(10)
+    >>> print(len(eigvals) - np.count_nonzero(eigvals)) #Count zero eigvalues
+    3
```
Comment (rossbar) [inline networkx/linalg/spectrum.py:50]: IMO the numpy operations actually make the example harder to understand. I'd vote to simply show the result:

```suggestion
    >>> nx.laplacian_spectrum(G)
    array([0., 0., 0., 2., 2.])
```

File: networkx/linalg/spectrum.py (hunk 2 above)
Comment (paulitapb) [reply networkx/linalg/spectrum.py]: true, thanks!!

**Review by rossbar — APPROVED** [review]
I took the liberty of removing the unused numpy import, but other than that LGTM. Thanks @paulitapb !

**Pull request merged**